scripts/publish.js

```diff
@@ -0,0 +1,117 @@
+const argv = require('yargs').argv
+const fs = require('fs');
+const fse = require('fs-extra');
+const util = require('util');
+const exec = require('child_process').exec;
+const semver = require('semver');
+const path = require('path');
+
+function panic(message) {
+    console.error(message);
+    process.exit(1);
+}
+
+async function execute(command, options = {}) {
+    const {doNotPanic, doNotLog} = options;
+    console.log(`execute command: ${command}`);
+    try {
+        return await (util.promisify(exec)(command, options).then(result => {
+            const {stdout, stderr} = result;
+            if(!doNotLog) {
+                console.log(stdout);
+                console.log(stderr);
+            }
+            return result;
+        }));
+    } catch (e) {
+        if (!doNotPanic)
+            panic(e);
+    }
+}
+
+async function run() {
+
+    let installedPackage = null;
+    let packageRoot = null;
+    const projectRoot = path.join(__dirname, '..');
+    const currentPackage = require(path.join(projectRoot, 'package.json'));
+    let isPreRelease = argv.prerelease;
+    let version = currentPackage.version;
+
+
+    // Networks must a comma separated list
+    if (!argv.networks || !argv.networks.split) {
+        panic("You must migrate to at least one network");
+    }
+    const networks = argv.networks.split(',');
+
+    // Git diff with a branch to detect if migration changed
+    if (argv.diffMigrationsFrom) {
+        const result = await execute(`git diff --name-only ${argv.diffMigrationsFrom}...HEAD`);
+        const migrationsChanged = result.stdout.split(/\r?\n/).filter(item => {
+            return /migrations\/\d+_(.*?).js$/.test(item)
+        });
+
+        if (migrationsChanged.length === 0) return;
+    }
+
+    // Check npm tag is set.
+    if (!argv.toTag || (typeof argv.toTag !== 'string')) {
+        panic("--to-tag must be set!")
+    }
+    // clean project build folder
+    await execute(`npm run clean`);
+
+    // If package is based on a previous published version
+    if (argv.fromTag) {
```

Comment:
will be a case where `fromTag` is not provided?

Reply:
If you want to do a new env you will not use fromTag.